### ISSUE TITLE
Glasovno upravljanje igrom Temple Run

### ISSUE THREAD

**Issue by stasadj** [ftn-ai-lab/sc-2021-siit#14]
## Tim:
SW-48/2018 Anastasija Đurić
SW-52/2018 Dina Petrov

## Asistent:
Dragan Vidaković

## Definicija problema:
Na osnovu data set-a koji se sastoji od audio sample-ova izgovorenih komandi za kretanje, generišu se odgovarajući spektogrami (MFCC) koji se koriste za obuku konvolucione neuronske mreže.
Obučeni model se potom koristi za mapiranje glasovnih komandi (koje korisnik zadaje putem mikrofona) na odgovarajuće tastere za upravljanje kretanjem u igri Temple Run u realnom vremenu.

## Skup podataka:
Skup podataka ćemo ručno sakupljati i sastojaće se od audio sample-ova izgovorenih komandi SKOČI, LEVO, DESNO, DOLE...
Nad svakim sakupljenim audio sample-om izvršiće se i audio manipulacije (pitch, echo, reverb, low/high pass filter...) kako bi se dataset dodatno proširio. 
Zbog mogućih ograničenja kod ručnog sakupljanja dataset-a u pogledu uspešnosti treniranja (nedovoljno širok obim uzoraka, overfitting), takođe ćemo treniranje vršiti i nad [postojećim skupom podataka](https://www.tensorflow.org/datasets/catalog/speech_commands) koji se sastoji od sličnih glasovnih komandi, ali na engleskom jeziku. Potom ćemo uporediti razlike u uspešnosti ova dva modela.

## Metodologija:
Ručno implementirana konvoluciona neuronska mreža

## Evaluacija: 
Dataset će biti podeljen na trening (70%), validacioni (20%) i test skup (10%).  Za metriku će biti koriščen procenat uspešno klasifikovanih komandi (accuracy). Sa obzirom na to da mnoge igre koriste iste komande za kretanje, rešenje je univerzano te se može testirati u realnom vremenu i nad drugim igrama osim Temple Run-a. 

## Repozitorijum:
[temple-run-voice-automation](https://github.com/PetrovDina/temple-run-voice-automation)

**Comment by vdragan1993** [comment]
Tema odobrena.
Srećan rad.
